File: C# Fundamentals/C# OOP Advanced/09.ReflectionN_Attributes/Ex/P03_BarraksWars/Core/Commands/RetireCommand.cs
namespace _03BarracksFactory.Core.Commands
{
    using _03BarracksFactory.Contracts;
    using _03BarracksFactory.Attributes;

    public class RetireCommand : Command
    {
        [Inject]
        private IRepository repository;

        public RetireCommand(string[] data) : base(data)
        {
        }

        public override string Execute()
        {
            string unitType = Data[1];
            this.repository.RemoveUnit(unitType);
            string output = unitType + " retired!";
            return output;
        }
    }
}

File: Java/01.Advanced/02.OOP/15.ReflectionAnnotation/Ex/src/barracksWars/core/commands/RetireCommand.java
package barracksWars.core.commands;

import barracksWars.annotations.Inject;
import barracksWars.interfaces.Repository;
import jdk.jshell.spi.ExecutionControl;

public class RetireCommand extends Command{

    @Inject
    private Repository repository;

    public RetireCommand(String[] data) {
        super(data);
    }

    @Override
    public String execute() throws ExecutionControl.NotImplementedException {
        String unitType = getData()[1];
        try {
            this.repository.removeUnit(unitType);
        } catch (IllegalArgumentException e) {
            return e.getMessage();
        }
        return getData()[1] + " retired!";
    }
}
